File: Polars.CSharp.Tests/TestUtils.cs
namespace Polars.CSharp.Tests;

public class DisposableFile : IDisposable
{
    public string Path { get; }

    public DisposableFile(string content, string extension = ".csv")
    {
        if (!extension.StartsWith(".")) extension = "." + extension;
        
        Path = System.IO.Path.Combine(System.IO.Path.GetTempPath(), $"{Guid.NewGuid()}{extension}");
        File.WriteAllText(Path, content);
    }

    public DisposableFile(string extension = ".parquet")
    {
        if (!extension.StartsWith(".")) extension = "." + extension;
        Path = System.IO.Path.Combine(System.IO.Path.GetTempPath(), $"{Guid.NewGuid()}{extension}");
    }

    public void Dispose()
    {
        if (File.Exists(Path))
        {
            try 
            {
                File.Delete(Path);
            }
            catch 
            {
            }
        }
    }
}

File: Polars.FSharp.Tests/TestUtils.fs
namespace Polars.FSharp.Tests

open System
open System.IO

type TempCsv(content: string) =
    let path = Path.GetTempFileName()
    do File.WriteAllText(path, content)
    
    member _.Path = path
    
    interface IDisposable with
        member _.Dispose() = 
            if File.Exists path then 
                try File.Delete path with _ -> () 